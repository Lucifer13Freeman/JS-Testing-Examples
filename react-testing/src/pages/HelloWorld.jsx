import { useState, useEffect } from 'react';

const HelloWorld = () => 
{
    const [data, setData] = useState(null);
    const [toggle, setToggle] = useState(false);
    const [value, setValue] = useState('');
    const [visible, setVisible] = useState(false);

    const onClick = () => 
    { 
        setToggle(prev => !prev); 
        value === 'hello' && setVisible(prev => !prev);
    }
    
    const onChange = (e) => setValue(e.target.value);

    useEffect(() => 
    {
        let timer = setTimeout(() => setData({}), 100);
        return () => clearTimeout(timer);
    }, []);

    return (
        <div>
            <h1 data-testid="value-elem">{value}</h1>
                <h1>hello world!</h1>
                { toggle === true && <div data-testid="toggle-elem" id="toggle-elem">toggle</div> }
                { data && <div style={{color: 'red'}}>data</div> }
                { visible && <h1 id="hello">HELLO WORLD!</h1> }
            <button data-testid="toggle-btn" id="toggle-btn" onClick={ onClick }>Click Me</button>
            <input onChange={ onChange } type="text" id="input-elem" placeholder="Input anything..."/>
        </div>
    );
};

export default HelloWorld;