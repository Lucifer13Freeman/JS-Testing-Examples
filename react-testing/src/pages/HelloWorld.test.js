import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
// import App from './App';
import HelloWorld from './HelloWorld';


describe('HELLO WORLD TEST', () => 
{
  test('test 1', () => 
  {
    render(<HelloWorld />);
  
    const helloWorldElement = screen.getByText(/hello world/i);
    const btn = screen.getByRole('button');
    const input = screen.getByPlaceholderText(/input anything/i);
  
    expect(helloWorldElement).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test('test 2', async () => 
  {
    render(<HelloWorld />);
    screen.debug();
    const helloWorldElement = await screen.findByText(/data/i);
    expect(helloWorldElement).toBeInTheDocument();
    expect(helloWorldElement).toHaveStyle({color: 'red'});
    screen.debug();
  });

  test("CLICK EVENT", () => 
  {
    render(<HelloWorld />);
    const btn = screen.getByTestId('toggle-btn');
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
    fireEvent.click(btn);
    expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument();
    fireEvent.click(btn);
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
  });

  test("INPUT EVENT", () => 
  {
    render(<HelloWorld />);
    const input = screen.getByPlaceholderText(/input anything/i);
    expect(screen.queryByTestId('value-elem')).toContainHTML('');
    userEvent.type(input, 'blabla');
    expect(screen.queryByTestId('value-elem')).toContainHTML('blabla');
  });
});