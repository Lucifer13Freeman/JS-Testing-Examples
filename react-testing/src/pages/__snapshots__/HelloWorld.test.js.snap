// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HELLO WORLD TEST test 1 1`] = `
<input
  id="input-elem"
  placeholder="Input anything..."
  type="text"
/>
`;
